### PR TITLE
Amp banner focus

### LINKED ISSUE #8895: Focus on AMP consent Banner not as expected
**Describe the bug**
AMP consent banner focus management does not meet a/c or screenreader UX 

**To Reproduce**
Steps to reproduce the behaviour:

1. View the banner (I used this 
2. Click ok on the privacy banner 
3. Note that the focus goes to the accept button (h2 banner heading is ignored) 
4. Click Manage my settings on the AMP consent banner 
5. Note that the focus doesn't shift to the h2 on the expanded banner 

**Expected behaviour**
[Focus management](https://paper.dropbox.com/doc/Amp-Consent-Banner-a11y-ac--BFYfVvus6LcbTIbciLWnU0hCAg-TVV52ZLGfomCNfvbmJ53f#:uid=299082894666340443569339&h2=Focus-management)
Scenario: Focus - Expanded banner 

- Given I have activated the “Manage my settings” button on the consent banner  
- When the expanded consent banner is displayed
- Then focus should be on the H2 at the top of banner e.g. (“Manage consent settings in AMP”) 

Scenario: Focus from buttons 

- Given I am on the consent banner 
- When I select to accept or reject data collection 
- Then the consent banner is closed  
- And my focus is returned to the last element that previously had focus on the page (ie. if the “ad choices / do not sell my info” link in the footer was my last focused element, then my focus should return here) 

**Screenreader UX**
Banner: https://zpl.io/VOZWqkj
Expanded banner: https://zpl.io/b6ql6ZR

**Screenshots**

**Devices and AT**

-iOS VoiceOver 
-MacOS VoiceOver 
-NVDA 
-JAWS
-Talkback on Android 

**Testing notes**

it is expected that we will need to retest this across our AT 
- [x] This bug fix is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by HarryVerhoef**
Resolves #8895

**Overall change:**
Fixes the incorrect focussing when using the AMP consent banner.

**Code changes:**

- Bumped psammead-brand to 7.3.0, where a `linkId` prop can be given.
- Refactor the header component and change props depending on whether or not `isAmp` is `true`.
- Use the AMP `.focus` actions on the `tap` event to pull focus where necessary.
- Where `.focus` cannot be used, `autoFocus` is used instead, to focus an element once it is not hidden.
- Updated snapshots

---

- [X] I have assigned myself to this PR and the corresponding issues
- [ ] I have added the `cross-team` label to this PR if it requires visibility across World Service teams
- [X] I have assigned this PR to the Simorgh project

**Testing:**

- [ ] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
- [X] This PR requires manual testing

### CODE REVIEW

File: src/app/containers/ConsentBanner/a11y.test.jsx [hunk 1]
```diff
@@ -75,3 +101,41 @@ describe('a11y', () => {
...
+        variant="default"
+      />,
+    );
+
+    const manageCookiesHeading = container.querySelector(
+      '#manageCookiesHeading',
+    );
+    manageCookiesHeading.focus();
+
+    expect(document.activeElement).toBe(manageCookiesHeading);
```
Comment (andrewscfc) [inline src/app/containers/ConsentBanner/a11y.test.jsx:120]: Were you not able to click the relevant buttons etc to test the focus behaviour as in the canonical test: https://github.com/bbc/simorgh/pull/8986/files#diff-84ca7f8699ea94dbc1f36ebb458987a2fcfbeebc9e6bf7e7a50808f1fa3bc904R99
?

**Review by andrewscfc — APPROVED** [review]

File: src/app/containers/ConsentBanner/a11y.test.jsx (hunk 1 above)
Comment (HarryVerhoef) [reply src/app/containers/ConsentBanner/a11y.test.jsx:120]: Unfortunately not, I tried but it seems that the AMP focus action doesn't work in these tests, even with the necessary scripts rendered

**Review by simonsinclair — APPROVED** [review]

**Comment by paruchurisilpa** [comment]
I have done a11y testing and it works as expected exactly like Canonical banner but the initial focus doesn't go to banner heading which is not an issue for AMP banner. LGTM..

**Comment by paruchurisilpa** [comment]
Completed a11y as mentioned on the spreadsheet here:

https://docs.google.com/spreadsheets/d/1Tw4NHHTecq1B5o5pcPQd395znjWZtRMD0YYhxCUQJBk/edit?usp=sharing

Moving to Ready to merge and Harry will review this with @greenc05 before merging.

**Review by mukama — APPROVED** [review]

**Pull request merged**